### ISSUE TITLE
Songs are saved with zero length

### ISSUE THREAD

**Issue by bitmarker** [jrnewell/spotify-ripper#8]
Me again :)

When ripping the playlist, some songs are saved with 00:00 time and also only about 40 kB. Removing the bad files and restarting the script brings better results.

´´´´
Setting artist: ++++++
Setting album: ++++++
Setting title: +++++++
Setting track info: (11, None)
Setting disc info: (1, 1)
Setting release year: 2015
Adding image cover.jpg
Time: 00:00     MPEG1, Layer III        [ ~159 kb/s @ 44100 Hz - Joint stereo ]
´´´´

Ripping with Raspberry Pi.

**Comment by jrnewell** [comment]
Do you have a spotify uri that I can test to reproduce that issue?  Off the top of my head I can do a check for 00:00 sec songs and skip over them.  Spotify is supposed to ignore unplayable tracks but I guess it is not foolproof.

**Comment by bitmarker** [comment]
The track is playable and is about 3 mins long. The script is playing the whole song, but seems to have some troubles when saving it to mp3.

**Comment by jrnewell** [comment]
ah ok, which track is it?

**Comment by bitmarker** [comment]
For example this playlist: spotify:user:118614068:playlist:1Qo24xMWid1WhoxFoUN18z

Some tracks are sometimes saved with time zero...

**Comment by jrnewell** [comment]
Do you remember if you saw the correct album/title before you saw that output with the ++++.  For example
```
Loading track...
Ripping spotify:track:0lOvtFFc7pNq2TsCohTCZt
/Users/james/python/spotify-ripper/Ray Charles/The Genius Hits The Road/Ray Charles - Georgia On My Mind.mp3
Progress: [========================================] 3:38 / 3:35
Rip complete
Ray Charles - Georgia On My Mind.mp3	[ 6.25 MB ]
```
It looks like an issue related to ```eyeD3``` (that is where that output comes from) unless the track somehow needed to reload its metadata again for some reason.  You said you can play that track with the issue to the end?

**Comment by bitmarker** [comment]
No, the track is saved with about 40 kB. It is not playing at all.

As far as I can remember, the track name was ok in those section. Only the time was 00:00.

**Comment by jrnewell** [comment]
ah ok, you didn't happen to see any errors or exceptions get logged before it quit ripping?  You said that it looked was ripping the whole song even though only 40 kB get saved?

**Comment by bitmarker** [comment]
Yes, the progress was running until the end of the time. Until it’s saving the file, everything looks like in cases where the song is saved ok.

I will run it again later and save the output so I can show it to you.

**Comment by jrnewell** [comment]
OK, thank you.

That is strange.  Lame should be streaming output to the encoded mp3 file as the data comes in.  I wonder if raspbian is doing a lot of file buffering because the disk is so slow.  I tried putting in a check to make sure the script waits for lame to finish before continuing.  Not sure if that will solve our problem here.

**Comment by bitmarker** [comment]
I've captured the output to the file:
http://filebin.ca/1yZH7GiD3Ifn/rip_log.txt

Track: spotify:user:spotify:playlist:0cpjMVagpkRrxLBhobs3uo
